src/Microsoft.Health.Dicom.Core/Features/Retrieve/RetrieveHelpers.cs

```diff
@@ -0,0 +1,33 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Features.Common;
+using Microsoft.Health.Dicom.Core.Features.Model;
+
+namespace Microsoft.Health.Dicom.Core.Features.Retrieve;
+public static class RetrieveHelpers
+{
+    public static async Task<FileProperties> CheckFileSize(IFileStore blobDataStore, long maxDicomFileSize, InstanceMetadata instance, CancellationToken cancellationToken)
+    {
+        EnsureArg.IsNotNull(blobDataStore, nameof(blobDataStore));
+        EnsureArg.IsNotNull(instance, nameof(instance));
+
+        FileProperties fileProperties = await blobDataStore.GetFilePropertiesAsync(instance.VersionedInstanceIdentifier.Version, cancellationToken);
+
+        // limit the file size that can be read in memory
+        if (fileProperties.ContentLength > maxDicomFileSize)
+        {
+            throw new NotAcceptableException(string.Format(CultureInfo.CurrentCulture, DicomCoreResource.RetrieveServiceFileTooBig, maxDicomFileSize));
```

Comment:
The error thrown to the user is
`Requested DICOM instance size is above the supported limit of {0}. The entire instance can be retrieved with the original transfer syntax by specifying 'transfer-syntax=*' in the Accept header`.

Do we want to throw the same error for rendered endpoint?

Reply:
updated error to mention render as well